docs/source/testing/mocking-responses.mdx

```diff
@@ -0,0 +1,181 @@
+
+To that effect, instead of mapping operations to responses with `register`, it is possible to **enqueue** responses:
+they will be returned in the order they were added. To do so, pass a `QueueTestNetworkTransportHandler` to
+the `TestNetworkTransport`, and use its `enqueue` function:
+
+```kotlin
+val handler = QueueTestNetworkTransportHandler()
+apolloClient = ApolloClient.Builder().networkTransport(TestNetworkTransport(handler)).build()
+
+handler.enqueue(query, testData1)
+handler.enqueue(query, testData2)
+handler.enqueue(query, testData3)
+
+val actual1 = apolloClient.query(query).execute().data
+val actual2 = apolloClient.query(query).execute().data
+val actual3 = apolloClient.query(query).execute().data
+```
+
+Note that you need to pass an operation to `enqueue` - this is because an `ApolloResponse` is built under the hood which
+has a reference to an operation. There is also a version of `enqueue` that takes only an `ApoolloResponse` as a
+parameter.
+
+Should you need more control over the responses to return, you can implement your own `TestNetworkTransportHandler`
+and pass it to the `TestNetworkTransport`:
+
+```kotlin
+class CustomTestNetworkTransportHandler : TestNetworkTransportHandler {
+  override fun handle(request: ApolloRequest<*>): ApolloResponse<*> {
+    return ApolloResponse.Builder(
+        /* Your custom logic here */
+    ).build()
+  }
+}
+
+apolloClient = ApolloClient.Builder()
+    .networkTransport(TestNetworkTransport(CustomTestNetworkTransportHandler()))
+    .build()
+```
+
+## `MockServer`
+
+While `TestNetworkTransport` functions at the GraphQL level, `MockServer` resides at the HTTP level and can be used to
+mock arbitrary HTTP responses.
+
+Add the dependency to your project's `build.gradle` file:
+
+```kotlin:title=build.gradle[.kts]
+dependencies {
+  testImplementation("com.apollographql.apollo3:apollo-mockserver:x.y.z")
+}
+```
+
+Here is how to use it:
+
+```kotlin
+// Create a mock server
+val mockServer = MockServer()
+
+// Use its URL on your ApolloClient
+val apolloClient = ApolloClient.Builder().serverUrl(mockServer.url()).store(store).build()
+
+// Enqueue responses
+mockServer.enqueue("""{"data": {"random": 42}}""")
+mockServer.enqueue(
+    MockResponse(
+        body = "Internal server error",
+        statusCode = 500,
+        headers = mapOf("X-Test" to "true"),
+        delayMillis = 1000L,
+    )
+)
+
+// Execute queries
+val response1 = apolloClient
+    .query(GetRandomQuery())
+    .execute()
+
+val response2 = apolloClient
+    .query(GetRandomQuery())
+    .execute()
+
+// Don't forget to stop the server when you're done
+mockServer.stop()
+```
+
+Note how the `enqueue` function normally takes a `MockResponse` as a parameter, while a shorthand version taking
+a `String` also exists.
+
+Also note how a delay can be passed to the `MockResponse` constructor to simulate network latency.
```

Comment:
Put this as comments in the sample code? By the time I reached this, I'm too lazy to scroll back 😅